src/linear_algebra/matrix.lean

```diff
@@ -813,6 +820,26 @@ begin
   apply h₂,
 end
 
+variables {R M : Type*} [comm_ring R] [add_comm_group M] [module R M]
+variables {b : n → M} (hb : is_basis R b)
+
+lemma to_lin_bijective [decidable_eq n] (A : matrix n n R) (hA : is_unit A.det) :
+  function.bijective (to_lin hb hb A) :=
+begin
+  refine function.bijective_iff_has_inverse.mpr ⟨to_lin hb hb A⁻¹, λ x, _, λ x, _⟩;
+    simp only [← linear_map.comp_apply, ← to_lin_mul,
+               matrix.nonsing_inv_mul _ hA, matrix.mul_nonsing_inv _ hA,
+               to_lin_one, linear_map.id_apply]
+end
```

Comment:
Why not the stronger statement with the shorter proof?
```suggestion
def to_lin_equiv [decidable_eq n] (A : matrix n n K) (hA : is_unit A.det) :
  V ≃ₗ[K] V :=
begin
  refine ⟨to_lin hb hb A, to_lin hb hb A⁻¹, λ x, _, λ x, _⟩;
    simp only [← linear_map.comp_apply, ← to_lin_mul,
               matrix.nonsing_inv_mul _ hA, matrix.mul_nonsing_inv _ hA,
               to_lin_one, linear_map.id_apply]
end
```

Reply:
I'm okay with doing so, but then we'd need to copy basically all of the `to_lin` API over to `to_lin_equiv`, which I want to avoid for now by just keeping it in terms of `to_lin`.

Reply:
I don't think you need any api other than `to_lin_equiv_apply`, which `@[simps]` will generate for you.